refactor: cleaned and updated AlphaCore Python & R implementations

prachics opened the pull request:
Hello Friedhelm, I am a UCF graduate student working with Dr. Akcora. I’ve cleaned up and refactored both the Python and R versions of AlphaCore, there are no changes to the core algorithm logic.
In this update we:

- Add an extractFeatures function that scans each node’s attributes for numeric values to use as features, falling back to computing in‑degree and strength if none are found.
- Add a safe fallback to a pseudo‑inverse when the covariance matrix is singular
- Remove the redundant computation of x_minus_center by computing it once and reusing its transpose, saving memory.

Please let me know if you’d like any further tweaks!

prachics committed:
refactor: cleaned and updated AlphaCore Python & R implementations

friedhelmvictor commented:
Hi Prachi!

Thank you for the effort!

I've tried this new version but it seems in contrast to the old version, this produces an error:

Running:
```
library(igraph)
library(data.table)
source("algorithms/alphaCore.R")

set.seed(1)

g <- erdos.renyi.game(200, 2/200, directed = T)
E(g)$weight <- 1:ecount(g)
V(g)$name <- paste("v", 1:vcount(g), sep="")

alphaCore(g)
```

used to yield:
```
     node     alpha batch
  1:   v1 0.0000000    18
  2:   v2 0.0000000    10
  3:   v3 0.0000000    12
  4:   v4 0.0000000     6
  5:   v5 0.4193455    24
 ---
196: v196 0.4193455    25
197: v197 0.0000000    12
198: v198 0.4193455    24
199: v199 0.0000000    11
200: v200 0.4193455    23
```

but with your changes yields:
```
Error in simple_vs_index(x, lazy_eval(args[[1]]), na_ok) :
  Unknown vertex selected
```

I haven't reviewed the python code yet.

Perhaps, what could be nice is to additionally add a test suite? So that it's easier to verify that the changes still lead to the same outcome?
Also, the examples in the readme would need to be updated.

prachics commented:
Hello Friedhelm,
Thanks for testing this so thoroughly!
I'll fix the R error, add a regression test suite, and update the README examples. I’ll keep you posted as soon as it’s fixed, let me know if you have any other suggestions in the meantime.

prachics committed:
Update README and alphaCore implementation

prachics committed:
Merge remote-tracking branch 'upstream/master' into update-alphaCore-implementations

prachics committed:
Add Python and R tests for AlphaCore with README validation

prachics commented:
Hello Friedhelm,

I've fixed the R implementation error and added  test suite for both Python and R versions. The tests validate that the output matches the expected structure and values from the README examples.

All implementations now produce expected results and the test suite ensures future changes maintain the same output. 

Please let me know if you'd like any further improvements!

prachics committed:
Add benchmark results

prachics committed:
Revert "Add benchmark results"

This reverts commit 5decf472f5d8f488b9c5af923b3e6c72a7422d37.

prachics committed:
Added Benchmark Results

prachics committed:
Revert "Added Benchmark Results"

This reverts commit 7bf1b1994d5f72542c46e4452f8f7df1488301a4.
